### DISCUSSION TITLE
Is it possible to access environment variables in templates?

### DISCUSSION THREAD

**Discussion by chopeen** [renovatebot/renovate#18763] [Request Help] [answered]
I use `renovate/renovate:latest` in my own GitLab instance. I would like to use `prFooter` to include a link to a GitLab job that created a particular MR. An easy way to do that would be to have a template like `This MR was created by Renovate bot, job URL is ${CI_JOB_URL}`.

I checked https://docs.renovatebot.com/templates/ and https://handlebarsjs.com/, but found no answer so far.

How can I access an env variable in templates? Or how can I pass it to Renovate before an MR is created?

---

One option I see is registering additional handler in [lib/util/template/index.ts](https://github.com/renovatebot/renovate/blob/main/lib/util/template/index.ts) that would expose the `process.env` list, but is it possible?

**Comment by viceice** [comment, marked as answer]
not possible

**Reply by chopeen** [reply]
Would you accept a PR with an new Handlebars handler (similar to https://github.com/renovatebot/renovate/pull/15212), to retrieve environment variables from `process.env`?

**Reply by viceice** [reply]
Please open a feature request first. we need to discuss some details first.

We can't simply allow accessing all env values for security reasons.

**Reply by chopeen** [reply]
OK, https://github.com/renovatebot/renovate/issues/18765.
